lib/AppInfo/Application.php

```diff
@@ -128,11 +128,22 @@ public function registerServices() {
 		$container->registerService('Crypt',
 			function (IAppContainer $c) {
 				$server = $c->getServer();
-				return new Crypt($server->getLogger(),
-					$server->getUserSession(),
-					$server->getConfig(),
-					$server->getL10N($c->getAppName()));
-			});
+				$hsmUrl = $this->config->getAppValue('encryption', 'hsm.url');
+				if (\is_string($hsmUrl) && $hsmUrl !== '') {
+					return new CryptHSM($server->getLogger(),
+						$server->getUserSession(),
+						$server->getConfig(),
+						$server->getL10N($c->getAppName()),
+						$server->getHTTPClientService(),
+						$server->getRequest(),
+						$c->query('OCP\AppFramework\Utility\ITimeFactory'));
+				} else {
+					return new Crypt($server->getLogger(),
+						$server->getUserSession(),
+						$server->getConfig(),
+						$server->getL10N($c->getAppName()));
+				}
+			})	;
```

Comment:
this semicolon is flying away :money_with_wings:

Reply:
Fixed.